Update plan file format docs

hamishwillee opened the pull request:
@DonLakeFlyer This moves the file format doc update discussed in https://github.com/mavlink/mavlink-devguide/pull/89/files to QGC. It is WIP - there will be comments inlines.

1. [x] Deleted the separate/old geofence/rally docs. 
1. You said that there is just one plan doc which includes rally, geofence and missions. I assume that even though these are uploaded together, the rally points and geofence will operate outside of a mission if they are triggered (e.g. by a failsafe)?
1. Other questions inline.

hamishwillee commented on en/file_formats/plan.md:
```diff
@@ -1,93 +1,143 @@
+`cruiseSpeed` | The default forward speed for Fixed wing or VTOL vehicles (i.e. when moving between waypoints).
+`hoverSpeed` | The default forward speed for multi-rotor vehicles.
+`items` | The list of mission item objects associated with the mission . The list may contain either/both [SimpleItem](#mission_simple_item) and [ComplexItem](#mission_complex_item) objects.
+`plannedHomePosition` | The planned home position is shown on the map and used for mission planning when no vehicle is connected. The array values shown above are (from top): latitude, longitude and AMSL altitude.
```
Is the altitude AMSL as I put?

DonLakeFlyer replied:
yup

hamishwillee commented on en/file_formats/plan.md:
```diff
@@ -1,93 +1,143 @@
-params | MISSION_ITEM.param1,2,3,4,x,y,z
+--- | ---
+`type` | `SimpleItem` for a simple item
+`AMSLAltAboveTerrain` | 
```
What are the next three values? AMSLAltAboveTerrain is a bit unclear because an alt is ususally either AMSL or above terrain, Altitude similarly could be either, and you have a mode, but often the altitude might depend on the frame. Can you confirm how this works.

DonLakeFlyer replied:
A mission item has 7 params with it. Sometimes called 1-7, sometimes 1-4, then x/y/z since th last three are usually used for lat/lon/alt but not always.

DonLakeFlyer replied:
AMSLAltAboveTerrain is just this value shown to the user:
![screen shot 2018-08-02 at 11 16 49 am](https://user-images.githubusercontent.com/5876851/43602713-9a7a65e2-9645-11e8-9440-56fddd854b80.png)

hamishwillee commented on en/file_formats/plan.md:
```diff
@@ -98,40 +148,246 @@ A complex item is a higher level encapsulation of multiple MISSION_ITEMS treated
-complexItemType | Specifies the type of complex item. QGroundControl currently supports the following types: [survey](../file_formats/survey.md), fwLandingPattern
-version | Specifies the version for this complex item.
+--- | ---
+`TransectStyleComplexItem` | ?
```
What is this? It seems to be a group for a particular type complex item, but then you ALSO have a type (survey). 
What are the other types of item?

DonLakeFlyer replied:
Survey and CorridorScan share a common base of information/code which is TransectStyleComplexItem.

hamishwillee commented on en/file_formats/plan.md:
```diff
@@ -98,40 +148,246 @@ A complex item is a higher level encapsulation of multiple MISSION_ITEMS treated
+`type` | `ComplexItem` for a complex item
+`complexItemType` | The type of complex item. *QGroundControl* currently supports the following types: [survey](../file_formats/survey.md), fwLandingPattern
+`version` | The version of the complex item specification. Here we document version 4.
+`angle` | 
```
Can you help explain what all the other fields are? Below this I included a cut down example (in HTML comments) that you might comment against.

hamishwillee replied:
Yes.

DonLakeFlyer replied:
* Angle is the grid in degrees
* Entry location is the entry point into the survey (0, 1, 2, 3). Which are the corners of the rect with encompasses the survey. They don't really correspond to specific corners. You just pick one and see if it is what you want.
* Fly alternate transects allows you to skip every other transect when flying and then come back and fly the ones you missed. Use for fixed wing when the turnaround is too acute such that the vehicle can't make the turn. Hence you fly every other which makes the turaround larger

DonLakeFlyer commented:
> You said that there is just one plan doc which includes rally, geofence and missions. I assume that even though these are uploaded together, the rally points and geofence will operate outside of a mission if they are triggered (e.g. by a failsafe)?

That is correct.

hamishwillee commented:
@DonLakeFlyer OK, I have:
- deleted the old rally and geofence docs
- Added new information including camera settings, survey, corridor and structure scans.

I will also delete the survey scan doc. 

1. Is there anything you want to capture from it?

This has lots of ? marking items I do not know. Most of the fields do not include units.

2. What is the best way to get them filled in? 
   - I push this to the main repo so you can edit directly?
   - Or should we just submit as is and you update the missing fields when you have time?
   - Something else?

3. This will be hard to maintain. It would be better if we generated docs on-the-fly. 
   - Does the system already contain docs for the fields/strings used?
   - if so, I know JSON does not support comments, but for the purpose of data export, might we allow developers to export a version with /** comments **/ after every field from source or field definitions? That would make it easier to keep in step with code.
   - Better ideas?

Essentially, I'm about as far as I can reasonably go with this for now. How do you want to proceed?

DonLakeFlyer commented:
> Or should we just submit as is and you update the missing fields when you have time?

I think that's the best choice.

> It would be better if we generated docs on-the-fly.

Would be nice, but not really possible given current coding.

hamishwillee commented:
Thanks @DonLakeFlyer . OK, merging. I removed the survey doc (now part of plan). As discussed, this is now ready for you to update fields etc, "as you will"
